lib/fizzy/execute.cpp

```diff
@@ -251,21 +254,50 @@ uint64_t eval_constant_expression(ConstantExpression expr,
         return globals[global_idx - imported_globals.size()];
 }
 
-void branch(uint32_t label_idx, LabelStack& labels, OperandStack& stack, const Instr*& pc,
+template <typename T>
+inline T read(const uint8_t*& input) noexcept
+{
+    T ret;
+    __builtin_memcpy(&ret, input, sizeof(ret));
+    input += sizeof(ret);
+    return ret;
+}
+
+void branch(const Code& code, LabelStack& labels, OperandStack& stack, const Instr*& pc,
     const uint8_t*& immediates) noexcept
 {
+    auto label_idx = read<uint32_t>(immediates);
+
     assert(labels.size() > label_idx);
     while (label_idx-- > 0)
         labels.pop();  // Drop skipped labels (does nothing for labelidx == 0).
     const auto label = labels.top();
     labels.pop();
 
-    pc = label.pc;
-    immediates = label.immediate;
+    const auto code_offset = read<uint32_t>(immediates);
+    const auto imm_offset = read<uint32_t>(immediates);
+    // TODO: these will be const when blocks other than loop use them
+    size_t stack_height = static_cast<uint32_t>(read<int>(immediates));
+    auto arity = read<uint8_t>(immediates);
```

Comment:
Is the reading happening here for the `label.pc == nullptr` just to skip ahead? Would it make sense using skip/read in the branches below?

Reply:
Yes, but these branches are removed in later commits and then it's used always